### ISSUE TITLE
Pack & Release version 2.5

### ISSUE THREAD

**Issue by OriRoth** [SpartanRefactoring/Spartanizer#333]
Hi everybody! We want to release version 2.5 soon, so we need to take care of some of the urgent issues.
I find it hard to follow all the different issues, so I'll post *here* some of the blockers that kill the plugin or prevent us from releasing in some other way.
Feel free to consult me. It is our **collective responsibility** to make this thing working.
Our main issues before releasing:
- [x] All tests pass ([0] failing)
- [x] Plugin working
  - LaconizeProject on...
    - [X] JUnit project
    - [x] Spartanizer project
- [x] All issues for version 2.5 are closed
  - https://github.com/SpartanRefactoring/Spartanizer/milestone/2

I am sorry but I have to delete some of your comments to keep this issue organized.

**Comment by dormaayan** [comment]
I disabled the tippers+tests that don't work (as @yossigil  told me to do), so now all the tests pass

**Comment by OriRoth** [comment]
Null pointer exception in `DeclarationAndForToFor#fitting` for:
```
int i;
for (i=0; i < MAX_PASSES; ++i) {
  final IProgressService ps=wb.getProgressService();
  final AtomicInteger passNum=new AtomicInteger(i + 1);
  final AtomicBoolean cancled=new AtomicBoolean(false);
  try {
    ps.run(true,true,pm -> {
      pm.beginTask("Spartanizing project '" + javaProject.getElementName() + "' - "+ "Pass "+ passNum.get()+ " out of maximum of "+ MAX_PASSES,us.size());
      int n=0;
      final List<ICompilationUnit> dead=new ArrayList<>();
      for (      final ICompilationUnit ¢ : us) {
        if (pm.isCanceled()) {
          cancled.set(true);
          break;
        }
        pm.worked(1);
        pm.subTask(¢.getElementName() + " " + ++n+ "/"+ us.size());
        if (!a.apply(¢))         dead.add(¢);
      }
      us.removeAll(dead);
      pm.done();
    }
);
  }
 catch (  final InvocationTargetException x) {
    LoggingManner.logEvaluationError(this,x);
  }
catch (  final InterruptedException x) {
    LoggingManner.logEvaluationError(this,x);
  }
  if (cancled.get() || us.isEmpty())   break;
}
```
in
```
return e.getType() == s.getType() && compareModifiers(step.extendedModifiers(e), step.extendedModifiers(s)) ? true : false;
```

**Comment by OriRoth** [comment]
Keep in mind ignored tests are not cool. Especially tests for active tippers.

**Comment by OriRoth** [comment]
`new BodyDeclarationModifiersSort.ofMethod()` appears twice in the `ToolBox` @dormaayan @sakopzon

**Comment by dormaayan** [comment]
@OriRoth : Those tippers are disabled now and Im not sure that they are correct.

**Comment by sakopzon** [comment]
For now there are two tests that fall due to collision of loop tippers with certification.
Another collision I know about is with inlining. This is the next step to fix.

**Comment by yossigil** [comment]
@sakopzon : I added few assertions to your code. Use them. Don't erase them.

**Comment by sakopzon** [comment]
OK

**Comment by OriRoth** [comment]
@yossigil current marker resolutions status: all resolutions seams to work. Notice "Laconize function" and "Laconize class" make only one pass.
Also: can I delete your comments from **this** issue? I want to keep it organized (this comment, for instance, will be deleted).

**Comment by sakopzon** [comment]
Is your nut pointer exception solved?
Didn't fix it directly because didn't manage to make from it a string that can be parsed, but since then fixed a lot of little bugs in different parts of the code. Can you check it?

**Comment by OriRoth** [comment]
No npe for Spartanizer project. You can check it for yourself by running the plugin on the project's code (or a copy of it, as I do). It seams most of the statements in the block of the for statement are irrelevant for your tipper, so you can create a test using:
```
trimming("" //
+ "int i;\n" //
+ "for (i=0; i < MAX_PASSES; ++i) {\n" //
+ "  if (cancled.get() || us.isEmpty())   break;\n" //
+ "}").to(...);
```
BTW now the plugin won't recommend any suggestions for this code.
- [ ] @sakopzon saw it

**Comment by sakopzon** [comment]
It shouldn't suggest, cause there is no statement that can be inserted to the for as I understand.

**Comment by OriRoth** [comment]
Version released. Conclusions:
* Updated version *not* available in eclipse marketplace - this is a problem from the last version release that needs to be investigated.
* Updated version available in https://github.com/SpartanRefactoring/Spartanizer/releases/tag/2.5.0
@sakopzon @yossigil @NivShalmon @dormaayan @dngreenst

**Comment by yossigil** [comment]
Done with 688f093dd44d25d014839ced8cd08466a7f55891
